File: PYTHON/Data Structures/Linear/Stack/Stack.py


stack = [0] * 5
top = 0

def Push(num):
    global top
    stack[top] = num
    top += 1

def Show():
    for i in range(len(stack)):
        print(stack[i])

def Pop():
    global top
    top -= 1

    num = stack[top]
    stack[top] = 0

    return num

def Peek():
    num = stack[top - 1]
    return num

File: JAVA/Data Structures/Linear/Stack/src/Stack.java
public class Stack
{
    int stack[] = new int[5];
    int top = 0;

    public void Push(int num)
    {
        stack[top] = num;
        top++;
    }

    public int Pop()
    {
        top--;
        int num = stack[top];

        stack[top] = 0;

        return num;
    }

    public int Peek()
    {
        int num = stack[top - 1];

        return num;
    }

    public void show()
    {
        for(int i = 0; i < stack.length; i++)
        {
            System.out.println(stack[i] + " ");
        }
    }

}
